src/integrations/__tests__/unixIntegrationManager.spec.ts

```diff
@@ -0,0 +1,300 @@
+  const srcPath = path.join(resourcesDir, 'kubectl');
+
+  await fs.promises.writeFile(dstPath, contents);
+  await manageSymlink(srcPath, dstPath, true);
+
+  const newContents = await fs.promises.readFile(dstPath, 'utf8');
+
+  expect(newContents).toEqual(contents);
+});
+
+test("manageSymlink should not touch the file if it isn't a symlink we own", async() => {
+  const oldSrcPath = path.join(testDir, 'fakeKubectl');
+  const dstPath = path.join(testDir, 'kubectl');
+  const srcPath = path.join(resourcesDir, 'kubectl');
+
+  await fs.promises.writeFile(oldSrcPath, 'contents');
+  await fs.promises.symlink(oldSrcPath, dstPath);
+  await manageSymlink(srcPath, dstPath, true);
+
+  const newTarget = await fs.promises.readlink(dstPath);
+
+  expect(newTarget).toEqual(oldSrcPath);
+});
+
+test("manageSymlink should not touch the file if custom string doesn't match", async() => {
+  const oldSrcPath = path.join(testDir, 'resources', os.platform(), 'bin', 'fakeKubectl');
+  const dstPath = path.join(testDir, 'kubectl');
+  const srcPath = path.join(resourcesDir, 'kubectl');
+
+  await fs.promises.symlink(oldSrcPath, dstPath);
+  await manageSymlink(srcPath, dstPath, true, path.join('another', 'dir'));
+
+  const newTarget = await fs.promises.readlink(dstPath);
+
+  expect(newTarget).toEqual(oldSrcPath);
+});
+
+test('manageSymlink should change the file if the custom string matches', async() => {
+  const customString = path.join('another', 'dir');
+  const oldSrcDir = path.join(testDir, customString);
+  const oldSrcPath = path.join(oldSrcDir, 'fakeKubectl');
+  const dstPath = path.join(testDir, 'kubectl');
+  const srcPath = path.join(resourcesDir, 'kubectl');
+
+  await fs.promises.mkdir(oldSrcDir, { recursive: true, mode: 0o755 });
+  await fs.promises.symlink(oldSrcPath, dstPath);
+  await manageSymlink(srcPath, dstPath, true, customString);
+
+  const newTarget = await fs.promises.readlink(dstPath);
+
+  expect(newTarget).toEqual(srcPath);
+});
+
+test('manageSymlink should delete the file if the target path matches', async() => {
+  const dstPath = path.join(testDir, 'kubectl');
+  const srcPath = path.join(resourcesDir, 'kubectl');
+
+  await fs.promises.symlink(srcPath, dstPath);
+  await manageSymlink(srcPath, dstPath, false);
+
+  return expect(fs.promises.readlink(dstPath)).rejects.toThrow();
+});
+
+test("manageSymlink shouldn't delete the file if the target path doesn't match", async() => {
+  const oldSrcPath = path.join(testDir, 'fakeKubectl');
+  const dstPath = path.join(testDir, 'kubectl');
+  const srcPath = path.join(resourcesDir, 'kubectl');
+
+  await fs.promises.writeFile(oldSrcPath, 'contents');
+  await fs.promises.symlink(oldSrcPath, dstPath);
+  await manageSymlink(srcPath, dstPath, false);
+
+  const newTarget = await fs.promises.readlink(dstPath);
+
+  expect(newTarget).toEqual(oldSrcPath);
+});
+
+test("manageSymlink shouldn't delete the file if it isn't a symlink", async() => {
```

Comment:
Same test as at line 176 `test("manageSymlink should not touch the file if it isn't a symlink", async() => {` ?